sequence_processing_pipeline/ConvertJob.py

```diff
@@ -210,3 +218,158 @@ def parse_job_script(job_script_path):
+                             "has completed.")
+
+        project_names = list(self.info.keys())
+
+        # confirm source project is a valid one.
+        if source_project not in project_names:
+            raise ValueError(f"'{source_project}' is not defined in the "
+                             "sample-sheet")
+
+        # confirm destination project is a valid one.
+        if dest_project not in project_names:
+            raise ValueError(f"'{dest_project}' is not defined in the "
+                             "sample-sheet")
+
+        if source_project == dest_project:
+            raise ValueError(f"source '{source_project}' and destination "
+                             f"'{dest_project}' projects are the same")
+
+        # note that the user can supply a sample-name that didn't make it
+        # through the conversion step and may have no files matched to it.
+        # this is considered okay and a possible outcome of conversion. In
+        # this case zero files are copied and no information is passed back
+        # to the user.
+
+        # projects that contain replicates must also be considered. if the
+        # value for sample_name contains a well-id, then only the files
+        # associated with that particular replicate should be copied. If the
+        # value instead references a sample_name in the 'orig_name' column,
+        # then all files associated with each replicate need to be moved.
+
+        # in this situation, the sample_name needs be compared against all
+        # orig_names in the project and the individual sample_names (w/well-
+        # ids) must be discovered. Then those individual sample_names can
+        # be processed.
+
+        if copy_all_replicates is True and self.contains_replicates is False:
+            raise ValueError("'treat_as_orig_name' is set to 'True' but this "
+                             "sample-sheet doesn't contain replicates")
+
+        samples = self.info[source_project]['samples']
+        sample_list = []
+
+        if copy_all_replicates:
+            for key in samples:
+                sample = samples[key]
+                # assume orig_name is present if treat_as_orig_name is True.
+                if sample_name == sample['orig_name']:
+                    sample_list.append(sample)
+        else:
+            # sample_name is a value from the sample_name column. it may or
+            # may not have a well-id appended and this sample-sheet may or
+            # may not contain replicates, but in either case a single sample
+            # either exists or it doesn't.
+            if sample_name in self.info[source_project]['samples']:
+                sample_list.append(samples[sample_name])
+
+        if len(sample_list) == 0:
+            # if the sample_list is empty, then sample-name wasn't present in
+            # either the sample_name or orig_name columns.
+            raise ValueError(f"'{sample_name}' is not defined in the project"
```

Comment:
I think the comment and message here are potentially misleading (although the message less so): for example, in a case where the user (mistakenly) entered an original sample name but said they wanted to copy all replicates for a prep that did have replicates, the entered value of `sample_name` *would* exist in the project--but it would be in the `orig_name` column from the [Data] whereas we would only have looked at the values in the `Sample_Name` column.  I mean, don't get me wrong, the user would still have entered an error-worthy combination of `sample_name` and `copy_all_samples`, but I would push for a more guaranteed-technically-accurate error message like:

`f"'{sample_name}' is not present as a '{name_col}' value in the project '{source_project}'"` (if we went with the approach suggested earlier)

Reply:
Actually if the user requested to copy all replicates and provided a value from the orig_name column then the sample would be in sample_list. This is because sample_list gets populated based on the values in the orig_name column or the sample_name column, depending if copy_all_replicates is True or False, respectively.

However I think I have a change that's in the spirit of what you're saying - we can differentiate the error message returned based on whether we were searching the orig_name or sample_name column. I also think sample_list is just a bad name - sample implying that the results are always based on sample_name column and Antonio dislikes variables that end in the type. :)

Reply:
What about samples? The this can be any kind of iterator: list, array, dict.keys(), etc.

Reply:
I'm going with 'results' since they are the results of the matching process. We use samples in another part of the code.